spec/query_builder_spec.cr

```diff
@@ -241,6 +241,27 @@ describe Avram::QueryBuilder do
       query.ordered?.should eq false
     end
   end
+
+  describe "clone" do
+    it "copies over all parts of a query" do
+      old_query = new_query
+        .select([:name, :age])
+        .join(Avram::Join::Inner.new(:users, :posts))
+        .where(Avram::Where::Equal.new(:name, "Paul"))
```

Comment:
Does it also copy raw_wheres? One more thing...I'm thinking we should automatically clone the query every time so that people don't even need to think  about it. It's just always a fresh copy. I think this could be done here in Queryable: 

Here's what it is now:

https://github.com/luckyframework/avram/blob/178e3e3dd282355e4d7b7b287bf0a70625ed0bbd/src/avram/queryable.cr#L41-L45

It could be now:

```
  def query
    # Don't reuse query, clone it
    if @query
      @query  = @query.clone
    else # If no query, create it
      @query = Avram::QueryBuilder
      .new(table: @@table_name)
      .select(@@schema_class.column_names)
     end
  end
```

Reply:
Yes, it does copy the raw_wheres because it calls `merge` first. I actually tried that bit from your branch you pushed up, but the specs wouldn't pass. I'll have to try again because I don't remember the error, but provided the specs are all correct, then I will get it to work.

Reply:
Ok awesome! And  yeah that was total untested code so maybe it won't work, but if we can that'd be awesome

Reply:
ok, yeah, this isn't working the way I'd hope. It's the first branch that gets me.

```crystal
# clone requires an arg
@query = @query.clone(what_would_go_here?)

# using `dup` instead causes nil errors all over.
@query = @query.dup
```

In the first one, we can't just pass the same query in because it wouldn't do anything. We'd have to create a new instance for it to work. That's what the `clone` method is already doing, so it's just double the work. Maybe you can see something I'm missing?